File: ObjectAvoidance/java/src/main/java/elite/math/trig/Cotangent.java
package elite.math.trig;

import elite.math.trig.triangle.Adjacent;
import elite.math.trig.triangle.Opposite;

/**
 *
 * @author CFerg (Elite)
 */
public class Cotangent {

    /**
     *
     */
    public double cotangent;
    
    /**
     *
     * @param a
     * @param o
     */
    public Cotangent(Adjacent a, Opposite o){
        this.cotangent = (a.adjacent / o.opposite);
    }
    
    /**
     *
     * @param arccot
     */
    public Cotangent(Arccotangent arccot){
        this.cotangent = (1 / arccot.arccotangent);
    }
    
    /**
     *
     * @param tan
     */
    public Cotangent(Tangent tan){
        this.cotangent = (1 / tan.tangent);
    }
}

File: ObjectAvoidance/src/elite/math/trig/Cotangent.kt
package elite.math.trig

import elite.math.trig.triangle.Adjacent
import elite.math.trig.triangle.Opposite

/**
 *
 * @author CFerg (Elite)
 */
class Cotangent {

    /**
     *
     */
    var cotangent: Double = 0.toDouble()

    /**
     *
     * @param a
     * @param o
     */
    constructor(a: Adjacent, o: Opposite) {
        this.cotangent = a.adjacent / o.opposite
    }

    /**
     *
     * @param arccot
     */
    constructor(arccot: Arccotangent) {
        this.cotangent = 1 / arccot.arccotangent
    }

    /**
     *
     * @param tan
     */
    constructor(tan: Tangent) {
        this.cotangent = 1 / tan.tangent
    }
}